src/jobManagerClient.ts

```diff
@@ -10,208 +10,186 @@ import {
   ITaskResponse,
   IUpdateJobBody,
   IUpdateTaskBody,
+  IFindJobsRequest,
 } from './models/dataTypes';
 import { httpClientConfig } from './models/utils';
 
 export class JobManagerClient extends HttpClient {
   public constructor(
     protected readonly logger: Logger,
     protected jobType: string,
-    protected taskType: string,
     protected jobManagerBaseUrl: string,
-    protected httpRetryConfig: IHttpRetryConfig = httpClientConfig
+    httpRetryConfig: IHttpRetryConfig | undefined = httpClientConfig,
+    targetService: string | undefined = 'jobManagerClient',
+    disableDebugLogs: boolean | undefined = true
   ) {
-    super(logger, jobManagerBaseUrl, 'jobManagerClient', httpRetryConfig, true);
+    super(logger, jobManagerBaseUrl, targetService, httpRetryConfig, disableDebugLogs);
   }
 
   public async getTask<T>(jobId: string, taskId: string): Promise<ITaskResponse<T> | null> {
+    const getTaskUrl = `/jobs/${jobId}/tasks/${taskId}`;
     try {
-      this.logger.info(`[JobManagerClient][getTask] jobId=${jobId}, taskId=${taskId}`);
-      const getTaskUrl = `/jobs/${jobId}/tasks/${taskId}`;
+      this.logger.debug({ jobId, taskId, url: getTaskUrl, targetService: this.targetService }, `getTask jobId=${jobId}, taskId=${taskId}`);
       const task = await this.get<ITaskResponse<T>>(getTaskUrl);
       return task;
     } catch (err) {
       this.logger.error(
-        `[JobManagerClient][getTask] jobId=${jobId}, taskId=${taskId} failed to get task error=${JSON.stringify(
-          err,
-          Object.getOwnPropertyNames(err)
-        )}`
+        { err, jobId, taskId, url: getTaskUrl, targetService: this.targetService },
```

Comment:
insert the message to msg attribute of the log object

Reply:
consider add msg string of error message parsing. in some case the entire error object printing is less readable for quiqe debugging and understanding